.storybook/config.js

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import styled from 'styled-components';
+import { configure, addDecorator } from '@storybook/react';
+import { withKnobs } from '@storybook/addon-knobs';
+import { withNotes } from '@storybook/addon-notes';
+import { withInfo } from '@storybook/addon-info';
+
+import { ThemeProvider } from 'styled-components';
+
+const StyledSelect = styled.select`
+  height: 50px;
+  width: 150px;
+`;
+
+const StyledContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+`;
+
+const SelectionContainer = styled.div`
+  width: 100%;
+  display: flex;
+  justify-content: flex-end;
+`;
+
+const ComponentContainer = styled.div``;
+
+
+
+class AppLayout extends React.Component {
+  state = {
+    activeTheme: 'theme1',
+    themes: {
+      theme1: {
+        main: 'palevioletred',
+        secondary: 'lightblue'
+      },
+      theme2: {
+        main: 'green',
+        secondary: 'lightblue'        
+      }
+
+    }
+  }
+
+  handleSelectChange = ({ target }) => this.setState({ activeTheme: target.value })
+
+  render() {
+    const { themes, activeTheme } = this.state;
+
+    return (
+      <StyledContainer>
+        <SelectionContainer>
+
+          <StyledSelect value={this.state.activeTheme} onChange={this.handleSelectChange}>
+            <option value='theme1'>theme1</option>
+            <option value='theme2'>theme2</option>
+          </StyledSelect>
+        </SelectionContainer>
+        <ComponentContainer>
+          <ThemeProvider theme={themes[activeTheme]}>{this.props.children}</ThemeProvider>
+        </ComponentContainer>
+      </StyledContainer>
+    )
+
+  }
+}
+
+const req = require.context('../src/components', true, /\.stories\.js$/);
+
+function loadStories() {
+  req.keys().forEach(filename => req(filename));
+}
+
+addDecorator(withKnobs);
+addDecorator(withNotes);
+addDecorator(withInfo);
+
```

Comment:
Are we sure that this global config is not adding unnecessary computation of the components that are being rendered. e.g. that the `withInfo` decorator is not running analysis on the other decorators, in addition to the Carbon components?

Reply:
Hmm... I tested it again and i think this addon doesn't work correctly for now.
I'll remove it for now and create separate task for it.